activesupport/lib/active_support/messages/rotator.rb

```diff
@@ -44,6 +45,8 @@ def build_rotation(secret = @secret, options)
 
       private
         def run_rotations(on_rotation)
+          on_rotation ||= @on_rotation
```

Comment:
This is a bit of a nit, but I find this can be a bit dangerous where callers of this method can get surprising behaviour if they pass in a `on_rotation` proc but another `@on_rotation` proc has been set.

A better approach could be something like passing the `@on_rotation` as the default instead of `nil`, in these two places: 

https://github.com/rails/rails/blob/af369f94e6834cb1b1f118ccf4e59b09373074e3/activesupport/lib/active_support/messages/rotator.rb#L36

https://github.com/rails/rails/blob/af369f94e6834cb1b1f118ccf4e59b09373074e3/activesupport/lib/active_support/messages/rotator.rb#L21

Since the `run_rotations` method can handle a `nil` value these should work 👌

Reply:
> if they pass in a on_rotation proc but another @on_rotation proc has been set.

The `on_rotation` proc passed when calling `decrypt_and_verify` takes precedence over the one passed in the constructor

Reply:
ah of course! my bad 🙇

Reply:
Thanks for pointing it out! I added a test to cover the case :)

Reply:
It would read cleaner to me if they were passed in as @itsWill said.